CMakeLists.txt

```diff
@@ -97,3 +97,11 @@ endif()
 if(RAJA_ENABLE_EXAMPLES)
   add_subdirectory(examples)
 endif()
+
+if(RAJA_ENABLE_APPLICATIONS)
+  if (NOT EXISTS ${PROJECT_SOURCE_DIR}/applications/CMakeLists.txt)
+    message(STATUS "Cloning RAJA example applications...")
+    execute_process(COMMAND git clone https://github.com/LLNL/RAJA-examples.git ${PROJECT_SOURCE_DIR}/applications)
```

Comment:
Do we want to leave the potential for additional repos to be incorporated? If so, perhaps the RAJA-examples checkout should live as a subdirectory within the applications directory. Else, 👍